src/WebJobs.Script/Utility.cs

```diff
@@ -879,6 +881,31 @@ public static bool IsResourceAzureBlobWithoutSas(Uri resourceUri)
             return true;
         }
 
+        public static string GetSasTokenExpirationDate(Uri resourceUri)
+        {
+            // Screen out URLs with an SAS token
+            var queryParams = HttpUtility.ParseQueryString(resourceUri.Query.ToLower());
+            if (queryParams != null && !string.IsNullOrEmpty(queryParams[SasTokenExpirationDate]))
+            {
+                return queryParams[SasTokenExpirationDate];
+            }
+
+            return string.Empty;
+        }
+
+        public static string GetSasTokenExpirationDateFromSasSignature(string azureWebJobsStorage)
+        {
+            var splitStorage = azureWebJobsStorage.Split(';');
```

Comment:
I think you can avoid the array allocation (`string.split` will allocate an array) with something like below:

```csharp
 var azureWebJobsStorageSpan = azureWebJobsStorage.AsSpan();
 var sasToken = azureWebJobsStorageSpan
                         .Slice(azureWebJobsStorageSpan.LastIndexOf(';') + 1).ToString();
```